### PR TITLE
[DEV-6609] Ensuring Fiscal Year for Account Data is Accurate in drop downs, tables, and Network Calls

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**High level description:**
This has been attempted by me twice. I made some mistakes and am now taking all of @jonhill13 's advice on his initial review of this.

-  Update redux to just have `submissionPeriods` from [the endpoint](https://api.usaspending.gov/api/v2/references/submission_periods/). I kept this in the `redux.account` key space because this data is strictly related to account data.
- In all the various places where account data is displayed, use redux as the source of truth and leverage existing unit-tested helpers to extract the following details: (a) the calendar date of the most recent submissions (this data is accurate as of...), (b) the latest Fiscal Year; (c) the latest fiscal period; (d) t he. latest fiscal quarter. 

**Technical details:**
This PR will be much easier to review if you reference the commits. There are three. Each are elaborate, specifying the technical details in the commit message:

bd39df4 Refactoring redux

9f2b692 Refactoring consumers of redux property to make use of new redux structure and leverage existing helpers

ab85bb5 Update tests to utilize correct props.

**JIRA Ticket:**
[DEV-6609](https://federal-spending-transparency.atlassian.net/browse/DEV-6609)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Provided instructions for testing in JIRA and PR `if applicable`
`N/A` Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
`N/A` Verified mobile/tablet/desktop/monitor responsiveness
`N/A` Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [x] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
`N/A` [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
`N/A` [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`
- [x] Smoke Tested in Sandbox
 
Reviewer(s):
`N/A` Design review complete `if applicable`
`N/A` [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete

### CODE REVIEW

File: src/js/components/aboutTheData/AboutTheDataPage.jsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ const AboutTheDataPage = ({
     history
 }) => {
     const { fy: urlFy, period: urlPeriod } = useParams();
-    const [dataAsOf, submissionPeriods] = useLatestAccountData();
+    const [, submissionPeriods, { year: latestFy, period: latestPeriod }] = useLatestAccountData();
```
Comment (ebdabbs) [inline src/js/components/aboutTheData/AboutTheDataPage.jsx:49]:
```suggestion
    const [submissionPeriods, { year: latestFy, period: latestPeriod }] = useLatestAccountData();
```

File: src/js/containers/account/AccountContainer.jsx [hunk 2]
```diff
@@ -25,10 +25,17 @@ import LoadingAccount from 'components/account/LoadingAccount';
 require('pages/account/accountPage.scss');
 
 const propTypes = {
-    currentFiscalYear: PropTypes.string,
+    latestSubmissionDate: PropTypes.string,
```
Comment (ebdabbs) [inline src/js/containers/account/AccountContainer.jsx:28]: Screen Shot 2021-01-13 at 1 40 46 PM
Looks like this prop is expected to be an object and duplicated below, nested under `latestPeriod`

File: src/js/components/agency/AgencyContent.jsx [hunk 3]
```diff
@@ -44,14 +43,14 @@ const agencySections = [
...
+    latestSubmissionDate: PropTypes.object
 };
 
 const AgencyContent = ({
     agency,
-    isTreasury,
-    dataAsOf
+    isTreasury
 }) => {
+    const [asOfDate, , { year: latestFy, quarter: latestQuarter }] = useLatestAccountData();
```
Comment (ebdabbs) [inline src/js/components/agency/AgencyContent.jsx:53]:
```suggestion
    const [asOfDate, { year: latestFy, quarter: latestQuarter }] = useLatestAccountData();
```

**Review by ebdabbs — CHANGES_REQUESTED** [review]
```Covid19Container.jsx:325  Error Submission Periods :  Object(...) is not a function``` seeing this error logged on the COVID profile

File: src/js/components/aboutTheData/AboutTheDataPage.jsx (hunk 1 above)
Comment (MaxwellKendall) [reply src/js/components/aboutTheData/AboutTheDataPage.jsx:49]: So to restructure an array, we have to do this I think

```javascript
   const [first, second, third] = [1,2,3]
   const [, second, third] = [1,2,3]
```

File: src/js/containers/covid19/homepage/CovidHighlights.jsx [hunk 4]
```diff
@@ -304,13 +307,14 @@ export class CovidHighlights extends React.Component {
             isDateLoading,
             hasNext
         } = this.state;
-        const { totalSpendingAmount, latestSubmissionPeriod } = this.props;
-        const parsedDate = latestSubmissionPeriod
-            ? latestSubmissionPeriod.format('MMMM DD[,] YYYY')
+        const { totalSpendingAmount, submissionPeriods } = this.props;
+        const parsedDate = submissionPeriods.length
+            ? getLatestPeriodAsMoment(submissionPeriods).format('MMMM DD[,] YYYY')
```
Comment (ebdabbs) [inline src/js/containers/covid19/homepage/CovidHighlights.jsx:312]: Should the homepage display January 8, 2021 with these changes? Is it based on `submission_reveal_date`? I see that submission in redux but it is still showing Nov 30.

File: src/js/helpers/accountHelper.js [hunk 5]
```diff
@@ -65,5 +49,45 @@ export const getSubmissionDeadlines = (fiscalYear, fiscalPeriod, submissionPerio
...
+            })
+            .map((s) => ({
+                revealDate: moment.utc(s.submission_reveal_date),
+                asOfDate: moment.utc(s.period_end_date),
+                period: s.submission_fiscal_month,
+                year: s.submission_fiscal_year,
+                quarter: s.submission_fiscal_quarter
+            }))
+            .sort(({ revealDate: a }, { revealDate: b }) => b.valueOf() - a.valueOf())
+            .find(({ revealDate: s }) => moment(s).isSameOrBefore(moment()));
```
Comment (ebdabbs) [inline src/js/helpers/accountHelper.js:69]: This looks correct but I'm still seeing Nov 30th on the homepage - should it show Jan 8 2021? That is the most recent reveal date I see in redux

File: src/js/components/aboutTheData/AboutTheDataPage.jsx (hunk 1 above)
Comment (ebdabbs) [reply src/js/components/aboutTheData/AboutTheDataPage.jsx:49]: Ah gotcha I was thinking it was leftover from a find and replace!

**Review by ebdabbs — APPROVED** [review]

**Pull request merged**